stl/inc/algorithm

```diff
@@ -4639,6 +4639,120 @@ _SampleIt sample(_PopIt _First, _PopIt _Last, _SampleIt _Dest, _Diff _Count,
 
     return _Dest;
 }
+
+#ifdef __cpp_lib_concepts
+namespace ranges {
+    // VARIABLE ranges::sample
+    class _Sample_fn : private _Not_quite_object {
+    public:
+        using _Not_quite_object::_Not_quite_object;
+
+        // clang-format off
+        template <input_iterator _It, sentinel_for<_It> _Se, weakly_incrementable _Out, class _Urng>
+            requires (forward_iterator<_It> || random_access_iterator<_Out>)
+                  && indirectly_copyable<_It, _Out> && uniform_random_bit_generator<remove_reference<_Urng>>

```

Comment:
`uniform_random_bit_generator` is declared in `<random>`.

I don't think we want to `#include <random>` here. Would it be a good idea to move `uniform_random_bit_generator` into `<xutility>`, next to `_Rng_from_urng`?

Reply:
> Would it be a good idea to move `uniform_random_bit_generator` into `<xutility>`

`<random>` includes `<algorithm>`, so we can simply push the concept definition up into `<algorithm>`. (And remove the unnecessary `#include <concepts>` from `<random>` as well.)